controllers/ibmpowervsmachine_controller_test.go

```diff
@@ -0,0 +1,562 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2/klogr"
+	"k8s.io/utils/pointer"
+	capiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	infrav1beta1 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cloud/scope"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/powervs/mock"
+)
+
+func TestIBMPowerVSMachineReconciler_Reconcile(t *testing.T) {
+	testCases := []struct {
+		name           string
+		powervsMachine *infrav1beta1.IBMPowerVSMachine
+		ownerMachine   *capiv1beta1.Machine
+		powervsCluster *infrav1beta1.IBMPowerVSCluster
+		ownerCluster   *capiv1beta1.Cluster
+		expectError    bool
+	}{
+		{
+			name:        "Should Reconcile successfully if no IBMPowerVSMachine found",
+			expectError: false,
+		},
+		{
+			name:           "Should Reconcile if Owner Reference is not set",
+			powervsMachine: &infrav1beta1.IBMPowerVSMachine{ObjectMeta: metav1.ObjectMeta{Name: "powervs-test-1"}, Spec: infrav1beta1.IBMPowerVSMachineSpec{ServiceInstanceID: "service-instance-1", Image: &infrav1beta1.IBMPowerVSResourceReference{}}},
```

Comment:
can you indent this struct as well for ease of read like below?

Reply:
and fix the indentation for all the test case structs as well.

Reply:
Fixed.